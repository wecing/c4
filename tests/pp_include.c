extern int printf(const char *s, ...);

#include "pp_include_x.h"

#define INCLUDE_Y "pp_include_y.h"
#include INCLUDE_Y

#define INCLUDE_Z "pp_include_z.h"
#define concat(a,b) a ## b
#include concat(INCLUDE, _Z)

// P is not defined.
#ifdef P
#include P
#endif

int main() {
  printf("X = %d\n", X);
  printf("Y = %d\n", Y);
  printf("Z = %d\n", Z);
  return 0;
}
