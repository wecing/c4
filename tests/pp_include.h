extern int printf(const char *s, ...);

#define X 42
