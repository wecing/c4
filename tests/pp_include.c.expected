X = 42
Y = 50
Z = 1024
